src/orchestrator/state.ts

```diff
@@ -23,37 +27,77 @@ export type AgentState = {
 
 export type ManifestEntry = {
   path: string;
-  test: any;
+  test: any; //eslint-disable-line @typescript-eslint/no-explicit-any
 }
 
 export type OrchestratorState = {
   agents: Record<string, AgentState>;
   manifest: ManifestEntry[];
 }
 
-export class State {
-  state: OrchestratorState = {
-    agents: {},
-    manifest: [],
-  }
+class Atom {
+  allocate: () => Operation;
+  update: (value) => Operation;
+  view: (lens) => Operation;
+  over: (lens, fn) => Operation;
+  get: () => Operation;
+  set: (lens, value) => Operation;
+  next: () => Operation;
 
-  get(): OrchestratorState {
-    return this.state;
-  }
+  constructor() {
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let id: string = Symbol('Atom<State>') as any;
+    let subscriptions = new EventEmitter();
 
-  update(fn: (OrchestratorState) => OrchestratorState): void {
-    this.state = fn(this.state);
-  }
+    let parent: Operation = ({ resume, context: { parent }}) => resume(parent.parent);
 
-  view(lens) {
-    return view(lens, this.get());
-  }
+    function* find(key: string): Operation {
+      for (let current: Context = yield parent; current; current = current.parent) {
+        if (current[key]) {
+          return current;
+        }
+      }
+      throw new Error('Atom not found. It must be allocated on the tree with the atom.allocate() operation');
+    }
 
-  set(lens, value) {
-    this.update((state) => set(lens, value, state) as unknown as OrchestratorState);
-  }
+    function* get(key: string) {
+      let context = yield find(key);
+      return context[key];
+    }
 
```

Comment:
This aspect of the atom; searching up the tree for the precise instance is not strictly necessary, but for something that represents the global state of the entire server I felt it was warranted. You can think of the atom like a "tree local" which is the effection equivalent of a "thread local" variable. You can still allocate an operation native object on the stack and pass it around like I did with the web socket `Client`. Here is an example using it to query state:

https://github.com/bigtestjs/server/blob/227b611f5bd47fa0f56350dabbdc541ce0796d8a/bin/query.ts